### PR TITLE
Fix #887: Update roadmap

### LINKED ISSUE #887: Roadmap update
**What happened**:
Since we have released v1.0, we should update the roadmap and features in release versions.
https://github.com/kubeedge/kubeedge#roadmap

**What you expected to happen**:
Correct roadmap in kubeedge

### PR DESCRIPTION
**Pull request by rohitsardesai83**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Update the roadmap with the 1.1 release milestone and future plan.

**Which issue(s) this PR fixes**:
Fixes #887 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by edisonxiang** [comment]
/lgtm
If it is needed, we can add chinese version in another pr.

File: docs/getting-started/roadmap.md [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+- Add Layer 4 proxy support in edgemesh.
+- Istio-based service mesh across Edge and Cloud where micro-services can communicate freely in the mesh.
 - Enable function as a service at the Edge.
-- Support more types of device protocols to Edge nodes such as AMQP, BlueTooth, ZigBee, etc.
+- Support more types of device protocols such as OPC-UA, Zigbee.
 - Evaluate and enable much larger scale Edge clusters with thousands of Edge nodes and millions of devices.
 - Enable intelligent scheduling of applications to large scale Edge clusters.
 - Data management/analytics framework.
-
-## Release 0.3
```
Comment (fisherxu) [inline docs/getting-started/roadmap.md]: Do we need to keep the previous things here?

File: docs/getting-started/roadmap.md [hunk 2]
```diff
@@ -1,20 +1,32 @@
 # Roadmap
 
-## Release 1.0 onwards
+This document defines a high level roadmap for KubeEdge development.
 
-KubeEdge will provide the fundamental infrastructure and basic functionality for IOT/Edge workloads. This includes: 
+The [milestones defined in GitHub](https://github.com/kubeedge/kubeedge/milestones) defined in GitHub represent the most up-to-date plans.
 
-- Istio-based service mesh across Edge and Cloud where micro-services can communicate freely in the mesh.
+KubeEdge will provide the fundamental infrastructure and basic functionality for IOT/Edge workloads.
```
Comment (fisherxu) [inline docs/getting-started/roadmap.md]: maybe no need to introduced at roadmap here :)

File: docs/getting-started/roadmap.md [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+- Replace data exchange format between cloud and edge from json to protobuf.
+- Support reliable message delivery from cloud to edge.
+- Evaluate gRPC for cloud to edge communication.
+- Support PV/PVC for stateful workloads at the edge.
+- Support ingress at edge.
 - Enhance performance and reliability of KubeEdge infrastructure.
+- Upgrade Kubernetes dependencies in vendor to v1.15.
+- Migrate to Go module for dependency management.
+- Improve contributor experience by defining project governance policies, release process, membership rules etc.
+- Improve the performance and e2e tests with more metrics and scenarios.
```
Comment (fisherxu) [inline docs/getting-started/roadmap.md]: Add the webhook for more validation to KubeEdge api object, xref

File: docs/getting-started/roadmap.md [hunk 4]
```diff
@@ -1,20 +1,32 @@
...
-- Istio-based service mesh across Edge and Cloud where micro-services can communicate freely in the mesh.
+KubeEdge will provide the fundamental infrastructure and basic functionality for IOT/Edge workloads.
+
+KubeEdge 1.0 is our current stable branch. The roadmap below outlines new features that will be added to KubeEdge.
+
+## Release 1.1
+- Replace data exchange format between cloud and edge from json to protobuf.
+- Support reliable message delivery from cloud to edge.
+- Evaluate gRPC for cloud to edge communication.
+- Support PV/PVC for stateful workloads at the edge.
```
Comment (fisherxu) [inline docs/getting-started/roadmap.md]: I think this is CSI integrated, not only for stateful workloads

File: docs/getting-started/roadmap.md (hunk 1 above)
Comment (rohitsardesai83) [reply docs/getting-started/roadmap.md]: I feel roadmap should list future work, so past release info need not be maintained. This info can be retrieved from the Github releases.

**Comment by rohitsardesai83** [comment]
@fisherxu , worked on review comments. please take a look. Thanks.

File: docs/getting-started/roadmap.md (hunk 3 above)
Comment (fisherxu) [reply docs/getting-started/roadmap.md]: Please help add the webhook to 1.1

Comment (rohitsardesai83) [reply docs/getting-started/roadmap.md]: its added at point 6. did I miss anything ?

Comment (fisherxu) [reply docs/getting-started/roadmap.md]: Sorry, missed it :)

File: docs/getting-started/roadmap.md [hunk 5]
```diff
@@ -1,20 +1,31 @@
...
+- Support CSI for persistent storage (using PV/PVC/StorageClass) at edge.
+- Support ingress at edge.
+- Add admission-webhook based validation for device CRDs.
 - Enhance performance and reliability of KubeEdge infrastructure.
+- Upgrade Kubernetes dependencies in vendor to v1.15.
+- Migrate to Go module for dependency management.
+- Improve contributor experience by defining project governance policies, release process, membership rules etc.
+- Improve the performance and e2e tests with more metrics and scenarios.
+
+## Future
```
Comment (fisherxu) [inline docs/getting-started/roadmap.md]: add the monitoring and telemetry for edge here

Comment (rohitsardesai83) [reply docs/getting-started/roadmap.md]: updated.

Comment (rohitsardesai83) [reply docs/getting-started/roadmap.md]: @fisherxu , updated, thanks.

**Comment by fisherxu** [comment]
Thanks @rohitsardesai83 
/lgtm
/cc @sids-b @kadisi @chendave

**Comment by sids-b** [comment]
Thanks @rohitsardesai83 

/approve

**Comment by fisherxu** [comment]
@sids-b need a lgtm label, others seems can't add it

**Comment by sids-b** [comment]
@fisherxu , I will add you guys as reviewers in OWNERS file.

Thanks @edisonxiang @fisherxu @subpathdev for helping review PRs

**Pull request merged**